### PR TITLE
Lf 2911 new abandonment date component for task abandonment flow

### PR DESCRIPTION
**Pull request by gursimran-singh**
**Description**

Rather than the standard date picker input allow user with following three options to select the date from:
1. The original due date 
- Followed by a pill showing the original due_date of the task
- Selected by default
2. Today
- Followed by a pill showing today’s date
3. Another date
- When selected, show…
- Label: “Which date?” 
- Input: a standard date picker selected and calendar open for selection

Jira link: https://lite-farm.atlassian.net/browse/LF-2911

**Type of change**

- [x] New feature (non-breaking change which adds functionality)

**How Has This Been Tested?**

- [x] UI components visually reviewed on desktop view
- [x] UI components visually reviewed on mobile view

**Checklist:**

- [x] The precommit and linting ran successfully
- [x] I have added or updated language tags for text that's part of the UI
- [x] I have added "MISSING" for all new language tags to languages I don't speak

### CODE REVIEW

File: packages/webapp/src/components/Task/AbandonTask/index.jsx
Comment (Duncan-Brain) [inline packages/webapp/src/components/Task/AbandonTask/index.jsx:24]: @gursimran-singh 

If you could put these in a constants.js file that would fit with the structure of the app. AssignTask would be a good place to check.

File: packages/webapp/src/components/Task/AbandonTask/index.jsx
Comment (Duncan-Brain) [inline packages/webapp/src/components/Task/AbandonTask/index.jsx:103]: @gursimran-singh 

If you could put these into a const above that makes the render more readable. See options in the CreateUserAccount component for examples.

File: packages/webapp/src/components/Task/AbandonTask/index.jsx
Comment (Duncan-Brain) [inline packages/webapp/src/components/Task/AbandonTask/index.jsx:173]: @gursimran-singh 

Forgot a console log :) haha

**Comment by Duncan-Brain** [comment]
@gursimran-singh 

It looks like things are working properly for the most part. A few things I am noticing:

1. Console errors not present in integration:
image

2. I think Mwayas ticket is in integration now so if you merge that you should pull in those features to be testable

3. Couple inline comments

File: packages/webapp/src/components/Task/AbandonTask/styles.module.scss
Comment (SayakaOno) [inline packages/webapp/src/components/Task/AbandonTask/styles.module.scss:11]: Do you have access to figma? https://www.figma.com/file/F3f6DpGcxAaEsxfl7OZkI2/Litefarm?node-id=30575%3A100361&t=r61M3p1gOPS6hZMy-0
The colors are darker than required. You can also check font-weight and font-size.

File: packages/webapp/src/components/Task/AbandonTask/styles.module.scss
Comment (SayakaOno) [inline packages/webapp/src/components/Task/AbandonTask/styles.module.scss:7]: Do you have access to figma? https://www.figma.com/file/F3f6DpGcxAaEsxfl7OZkI2/Litefarm?node-id=30575%3A100361&t=r61M3p1gOPS6hZMy-0
You can check font-weight and font-size there.

Comment (JimmyRowland) [reply packages/webapp/src/components/Task/AbandonTask/styles.module.scss:7]: Also storybook
https://5fa09c09b8435e0022120d64-apnfscolio.chromatic.com/?path=/story/components-typography--override
https://5fa09c09b8435e0022120d64-apnfscolio.chromatic.com/?path=/story/components-filterpill--selected

Comment (gursimran-singh) [reply packages/webapp/src/components/Task/AbandonTask/styles.module.scss]: @JimmyRowland  I tried to use Pill component for this but got following error
Screenshot 2023-02-18 at 7 14 55 AM

RadioGroup uses P tag to display its label and Pill uses div and we should not have div inside p. This needs a separate discussion on properly using of tags to display text inside components.

Comment (JimmyRowland) [reply packages/webapp/src/components/Task/AbandonTask/styles.module.scss]: You can update radio/radioGroup/pill components to support general patterns like these.

If this date pill pattern only appears here, the pill component can be passed as a prop to each option.

Comment (gursimran-singh) [reply packages/webapp/src/components/Task/AbandonTask/styles.module.scss]: Yes, we can, but i think this would be out of scope for this ticket. This issue needs a separate ticket to be addressed.

**Review by Duncan-Brain — DISMISSED** [review]
This works for me and I no longer see the console errors mentioned previously.

File: packages/webapp/src/components/Task/AbandonTask/styles.module.scss
Comment (SayakaOno) [reply packages/webapp/src/components/Task/AbandonTask/styles.module.scss]: Created tickets:
- https://lite-farm.atlassian.net/browse/LF-2996
- https://lite-farm.atlassian.net/browse/LF-2997

File: packages/webapp/src/components/Task/AbandonTask/styles.module.scss
Comment (gursimran-singh) [reply packages/webapp/src/components/Task/AbandonTask/styles.module.scss:16]: I think the current css will not be used if we planning to use Pill component. let's first implement that and if needed we can change the css accordingly.

Comment (kcussen) [reply packages/webapp/src/components/Task/AbandonTask/styles.module.scss:16]: @gursimran-singh thanks so much for making this PR! Would you mind making the in line changes Sayaka has suggested so we can get the functionality in during this sprint? We've got tickets in next sprint (LF-2996 and LF-2997) to address your tech debt concerns.

Comment (gursimran-singh) [reply packages/webapp/src/components/Task/AbandonTask/styles.module.scss:16]: Made the needed Css changes

File: packages/webapp/src/components/Task/AbandonTask/styles.module.scss
Comment (SayakaOno) [inline packages/webapp/src/components/Task/AbandonTask/styles.module.scss:25]: Thank you for adding these.
I don't think line-height is working as you intended. I think one way to make it work is to set `display: inline-block;` and adjust its padding.

Comment (gursimran-singh) [reply packages/webapp/src/components/Task/AbandonTask/styles.module.scss:25]: Done the changes

File: packages/webapp/src/components/Task/AbandonTask/styles.module.scss
Comment (kcussen) [reply packages/webapp/src/components/Task/AbandonTask/styles.module.scss:16]: 🙇

File: packages/webapp/src/components/Task/AbandonTask/styles.module.scss
Comment (SayakaOno) [reply packages/webapp/src/components/Task/AbandonTask/styles.module.scss:25]: It looks like no padding top and button are needed...
I think this is a good time to check in with Carolina and get an approval from her, if you have not.

Comment (gursimran-singh) [reply packages/webapp/src/components/Task/AbandonTask/styles.module.scss:25]: Done, and i have requested Carolina to review it.

Comment (SayakaOno) [reply packages/webapp/src/components/Task/AbandonTask/styles.module.scss:25]: Carolina mentioned that she would review it in beta, so I will go ahead and merge. (I personally think that it would be more efficient to share a screenshot with Carolina and get an approval before merging a PR in case she has a request. Sorry for lack of clarification)
Thank you @gursimran-singh!

**Review by SayakaOno — APPROVED** [review]

**Pull request merged**
